tests/integration/ha/test_ha.py

```diff
@@ -0,0 +1,321 @@
+
+    # Un-freeze the process
+    logger.info(f"Un-freezing broker: {initial_leader_num}")
+    await send_control_signal(
+        ops_test=ops_test, unit_name=f"{APP_NAME}/{initial_leader_num}", signal="SIGCONT"
+    )
+    await asyncio.sleep(REELECTION_TIME)
+    topic_description = get_topic_description(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+
+    # verify the unit is now rejoined the cluster
+    assert topic_description.in_sync_replicas == {0, 1, 2}
+
+    result = c_writes.stop()
+    assert_continuous_writes_consistency(result=result)
+
+
+async def test_full_cluster_crash(
+    ops_test: OpsTest,
+    restart_delay,
+    c_writes: ContinuousWrites,
+    c_writes_runner: ContinuousWrites,
+):
+    # Let some time pass for messages to be produced
+    await asyncio.sleep(10)
+
+    logger.info("Killing all brokers...")
+    # kill all units "simultaneously"
+    await asyncio.gather(
+        *[
+            send_control_signal(ops_test, unit.name, signal="SIGKILL")
+            for unit in ops_test.model.applications[APP_NAME].units
+        ]
+    )
+    # Give time for the service to restart
+    await asyncio.sleep(RESTART_DELAY * 2)
+
+    topic_description = get_topic_description(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+    initial_offsets = get_topic_offsets(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+    await asyncio.sleep(CLIENT_TIMEOUT * 2)
+    next_offsets = get_topic_offsets(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+
+    assert int(next_offsets[-1]) > int(initial_offsets[-1])
+    assert topic_description.in_sync_replicas == {0, 1, 2}
+
+    result = c_writes.stop()
+    assert_continuous_writes_consistency(result=result)
+
+
+async def test_full_cluster_restart(
+    ops_test: OpsTest,
+    c_writes: ContinuousWrites,
+    c_writes_runner: ContinuousWrites,
+):
+    # Let some time pass for messages to be produced
+    await asyncio.sleep(10)
+
+    logger.info("Restarting all brokers...")
+    # Restart all units "simultaneously"
+    await asyncio.gather(
+        *[
+            send_control_signal(ops_test, unit.name, signal="SIGTERM")
+            for unit in ops_test.model.applications[APP_NAME].units
+        ]
+    )
+    # Give time for the service to restart
+    await asyncio.sleep(REELECTION_TIME * 2)
+
+    initial_offsets = get_topic_offsets(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+    await asyncio.sleep(CLIENT_TIMEOUT * 2)
+    next_offsets = get_topic_offsets(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+    topic_description = get_topic_description(ops_test=ops_test, topic=ContinuousWrites.TOPIC_NAME)
+
+    assert int(next_offsets[-1]) > int(initial_offsets[-1])
+    assert topic_description.in_sync_replicas == {0, 1, 2}
+
+    result = c_writes.stop()
+    assert_continuous_writes_consistency(result=result)
+
+
+# @pytest.mark.skip
+# async def test_multi_cluster_isolation(ops_test: OpsTest, kafka_charm):
```

Comment:
why is this commented out or skipped?

Reply:
Ah, thanks. Need to re-add that to the tests.